### ISSUE TITLE
Multi select dropdown multiSelectItem missing role='option'.

### ISSUE THREAD

**Issue by vinodneppalli62** [primefaces/primeng#14958]
### Describe the bug

1. JAWS screen reader reading selected checkbox as not selected while navigating through multi select dropdown options.
2. Select All checkbox is always reading checkbox checked whether is checked or not (I think instead of setting false value to checked attribute on 'checkbox input' removing that attribute when it is not checked works).

### Environment

Using prime ng multiselect in the application currently we are upgrading. We are using 16.9.1 and It is happening in latest version as well.

### Angular version

^16.2.0

### PrimeNG version

^16.9.1 

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.9.0

### Browser(s)

Chrome

### Expected behavior

Should read checked one as Selected and not checked one as Not Selected.

**Comment by vinodneppalli62** [comment]
role='option' fixed the options accessibility but select all check box near filter input is always reading checked even though it is not selected.
